misk/src/main/kotlin/misk/clustering/fake/FakeCluster.kt

```diff
@@ -0,0 +1,73 @@
+package misk.clustering.fake
+
+import com.google.common.util.concurrent.Service
+import com.google.inject.Key
+import misk.DependentService
+import misk.clustering.Cluster
+import misk.clustering.ClusterHashRing
+import misk.clustering.ClusterResourceMapper
+import misk.clustering.ClusterWatch
+import misk.clustering.DefaultCluster
+import java.util.concurrent.Executor
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+/**
+ * A [FakeCluster] is a [Cluster] that is a fake. It delegates entirely to [DefaultCluster],
+ * but allows us to keep [DefaultCluster] internal and make it clear that [FakeCluster]
+ * is only intended to be used for fakes.
+ *
+ * NB(mmihic): I'd prefer for this to be in the misk-testing module, but it can't since it
+ * relies on [DefaultCluster] and we want to leave [DefaultCluster] client
+ */
+@Singleton
+class FakeCluster(
+  self: Cluster.Member = Cluster.Member(SELF_NAME, SELF_IP),
+  newResourceMapperFn: (Set<Cluster.Member>) -> ClusterResourceMapper = { ClusterHashRing(it) }
+) : Cluster, Service, DependentService {
```

Comment:
does something break implementation delegation here?
https://kotlinlang.org/docs/reference/delegation.html#implementation-by-delegation

Reply:
nope, just wasn't aware of this. nice feature

Reply:
Ah ok so it requires that the delegate be a constructor parameter, but I don't want to expose that type externally, so won't work here

Reply:
eh i made it work